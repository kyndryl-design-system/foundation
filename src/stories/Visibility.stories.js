import { html } from 'lit';

export default {
  title: 'Foundation/Visibility',
  decorators: [
    (story) =>
      html`
        <style>
          .storybook-visibility-example {
            align-items: center;
            background: var(--kd-color-cloud-10);
            display: flex;
            margin: 16px;
            min-height: 4em;
            padding: 16px;
          }
        </style>
        ${story()}
      `,
  ],
};

const args = {};

export const Visibility = {
  args: args,
  render: () => {
    return html`
      <div class="storybook-visibility-example">
        This block includes text for screen readers only.
        <div class="kd-visibility--sr-only">screen reader only content</div>
      </div>
      <div class="storybook-visibility-example">
        <div class="kd-visibility--hidden-sm">
          Text hidden at Small (sm) breakpoint.
        </div>
      </div>
      <div class="storybook-visibility-example">
        <div class="kd-visibility--hidden-md">
          Text hidden at Medium (md) breakpoint.
        </div>
      </div>
      <div class="storybook-visibility-example">
        <div class="kd-visibility--hidden-lg">
          Text hidden at Large (lg) breakpoint.
        </div>
      </div>
      <div class="storybook-visibility-example">
        <div class="kd-visibility--hidden-xl">
          Text hidden at X-Large (xl) breakpoint.
        </div>
      </div>
      <div class="storybook-visibility-example">
        <div class="kd-visibility--hidden-max">
          Text hidden at Max (max) breakpoint.
        </div>
      </div>
    `;
  },
};
