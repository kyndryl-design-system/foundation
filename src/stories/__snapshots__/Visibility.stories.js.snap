// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foundation/Visibility Visibility smoke-test 1`] = `
<style>
  .storybook-visibility-example {
            align-items: center;
            background: var(--kd-color-cloud-10);
            display: flex;
            margin: 16px;
            min-height: 4em;
            padding: 16px;
          }
</style>
<div class="storybook-visibility-example">
  This block includes text for screen readers only.
  <div class="kd-visibility--sr-only">
    screen reader only content
  </div>
</div>
<div class="storybook-visibility-example">
  <div class="kd-visibility--hidden-sm">
    Text hidden at Small (sm) breakpoint.
  </div>
</div>
<div class="storybook-visibility-example">
  <div class="kd-visibility--hidden-md">
    Text hidden at Medium (md) breakpoint.
  </div>
</div>
<div class="storybook-visibility-example">
  <div class="kd-visibility--hidden-lg">
    Text hidden at Large (lg) breakpoint.
  </div>
</div>
<div class="storybook-visibility-example">
  <div class="kd-visibility--hidden-xl">
    Text hidden at X-Large (xl) breakpoint.
  </div>
</div>
<div class="storybook-visibility-example">
  <div class="kd-visibility--hidden-max">
    Text hidden at Max (max) breakpoint.
  </div>
</div>
`;
